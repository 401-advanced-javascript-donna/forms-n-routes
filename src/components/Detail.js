import React, { Component } from 'react';
import PropTypes from 'prop-types';

class CharacterDetail extends Component {

  static propTypes = {
    match: PropTypes.shape({
      params: PropTypes.shape({
        name: PropTypes.string.isRequired
      }).isRequired
    }).isRequired
  };

  state = {
    character: []
  }


  componentDidMount() {
    getSingleCharacter(this.props.match.params.name)
      .then(character => {
        this.setState({ character });
      });
  }


  render() {

    return (
      <div>
        <h1>{this.state.character.name}</h1>
        <img src={this.state.character.photoUrl}></img>
        <p>Gender: {this.state.character.gender}</p>
        <p>Hair: {this.state.character.hair}</p>
        <p>Affiliation: {this.state.character.affiliation}</p>
      </div>
    );
  }

}


export default CharacterDetail;
