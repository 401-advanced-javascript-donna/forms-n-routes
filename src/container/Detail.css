.Detail {
  width: 60vw;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 0 auto 100px auto;

  h1 {
    text-align: center;
    color: black;
    font-weight: bold;
    font-size: 1.2em;
    text-transform: uppercase;
    font-weight: 700;
    font-family: sans-serif;
    margin-top: 35px;
  }

  img {
    width: 40%;
    border: 4px solid #A0B890;
    margin-top: 20px;
    box-shadow: 3px 1px 5px black;
  }


  p {
    text-align: center;
    color: black;
    text-decoration: none;
    font-family: sans-serif;
    margin: 15px;
  }
}