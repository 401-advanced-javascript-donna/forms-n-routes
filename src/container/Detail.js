import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { getSingleCharacter } from '../services/api-call';
import styles from './Detail.css';

class Detail extends Component {

  static propTypes = {
    match: PropTypes.shape({
      params: PropTypes.shape({
        name: PropTypes.string.isRequired
      }).isRequired
    }).isRequired
  };

  state = {
    character: []
  }


  componentDidMount() {
    getSingleCharacter(this.props.match.params.name)
      .then(character => {
        this.setState({ character });
      });
  }


  render() {

    return (
      <div className={styles.Detail}>
        <h1>{this.state.character.name}</h1>
        <img src={this.state.character.photoUrl}></img>
        <p>Gender: {this.state.character.gender}</p>
        <p>Hair: {this.state.character.hair}</p>
        <p>Affiliation: {this.state.character.affiliation}</p>
      </div>
    );
  }

}


export default Detail;
