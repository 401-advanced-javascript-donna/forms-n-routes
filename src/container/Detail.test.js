import React from 'react';
import { shallow } from 'enzyme';
import Detail from './Detail';

jest.mock('../services/api-call', () => ({
  getSingleCharacter() {
    return Promise.resolve([
      {
        '_id': '5cf5679a915ecad153ab68d3',
        'name': 'Appa',
        'photoUrl': 'https://vignette.wikia.nocookie.net/avatar/images/6/65/Appa_flying.png/revision/latest?cb=20140517110636',
        'gender': 'male'
      }
    ]);
  }
}));


describe('Detail component', () => {

  it('renders', () => {
    const wrapper = shallow(
      <Detail
        match={{
          params: {
            name: 'arnold'
          }
        }} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
