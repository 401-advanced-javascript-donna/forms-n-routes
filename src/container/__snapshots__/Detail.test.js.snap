// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail component renders 1`] = `
<div>
  <h1 />
  <img />
  <p>
    Gender: 
  </p>
  <p>
    Hair: 
  </p>
  <p>
    Affiliation: 
  </p>
</div>
`;
